packages/components/plh/plh-kids-kw/components/module-list-item/module-list-item.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, OnInit } from "@angular/core";
+import { FlowTypes } from "packages/data-models";
+import { TemplateBaseComponent } from "src/app/shared/components/template/components/base";
+import { TemplateTranslateService } from "src/app/shared/components/template/services/template-translate.service";
+import {
+  getBooleanParamFromTemplateRow,
+  getStringParamFromTemplateRow,
+} from "src/app/shared/utils";
+
+interface IModuleListItemParams {
+  /* TEMPLATE PARAMETER: "module_image_asset". The image attached to the module */
+  moduleImageAsset: string | null;
+  /* TEMPLATE PARAMETER: "text_transform". The format of the text on the module item. Default null */
+  textTransform: "capitalise" | "uppercase" | null;
+  /* TEMPLATE PARAMETER: "is_locked". The boolean that marks the module as locked or unlocked */
+  isLocked: boolean;
+  /* TEMPLATE PARAMETER: "nav_image_asset". The navigation icon*/
+  navImageAsset: string | null;
+  /* TEMPLATE PARAMETER: "locked_image_asset". The locked icon*/
+  lockedImageAsset: string | null;
```

Comment:
nit(non-blocking): assuming the author doesn't provide these variables they will come through as `undefined` (and not `null`). Makes no real semantic difference, but might be easier to author the interface in the same way by marking optional properties with a `?`, e.g.
```ts
interface IModuleListItemParams{

lockedImageAsset?: string
}
```